src/identity.cc

```diff
@@ -1114,12 +957,6 @@ TRITONBACKEND_ModelInstanceExecute(
     SET_TIMESTAMP(compute_end_ns);
     max_compute_end_ns = compute_end_ns;
 
-#ifdef TRITON_ENABLE_GPU
-    if (cuda_copy) {
-      cudaStreamSynchronize(instance_state->CudaStream());
-    }
-#endif  // TRITON_ENABLE_GPU
```

Comment:
You still need to synchronize before sending the response when CUDA copy is used, otherwise the PR looks good to me